File: frontend/src/branding.js
/**
 * The practice's identity, in one place.
 *
 * Mirrors branding.py on the backend, which owns the copy printed on referral
 * PDFs. Kept as one module so renaming the practice is a single edit rather
 * than a hunt through the rail, the login screen and the page title.
 *
 * VITE_PRACTICE_NAME overrides it at build time, matching the backend's
 * PRACTICE_NAME environment variable, so a second practice deploying this needs
 * no code change.
 */
const DEFAULT_PRACTICE_NAME = 'Passion Dental';

/** A blank override is a misconfiguration, not a request for no name. */
export const PRACTICE_NAME =
  import.meta.env?.VITE_PRACTICE_NAME?.trim() || DEFAULT_PRACTICE_NAME;

/** What the product does, shown under the name on the sign-in card. */
export const PRACTICE_TAGLINE = 'Third molar review portal';

/**
 * The monogram on the rail and sign-in tiles: first letters of the first two
 * words, so "Passion Dental" reads as PD rather than a bare P.
 */
export const PRACTICE_INITIALS = PRACTICE_NAME
  .split(/\s+/)
  .filter(Boolean)
  .slice(0, 2)
  .map((word) => word[0])
  .join('')
  .toUpperCase();

File: branding.py
"""
branding.py - The practice's identity, in one place.

The practice name is printed on referral PDFs that go to patients, so it must
have exactly one definition; a literal retyped at each call site is a name that
eventually disagrees with itself.

`PRACTICE_NAME` is the constant for import-time use. `practice_name()` reads the
environment on every call so a deployment can override it without a code change,
which is what a second practice running this needs.
"""
import os

DEFAULT_PRACTICE_NAME = "Passion Dental"

#: Tagline under the heading on the referral PDF.
PRACTICE_TAGLINE = "panoramic radiograph review"


def practice_name() -> str:
    """The practice name, honouring a PRACTICE_NAME override.

    A blank or whitespace-only override is a misconfiguration rather than a
    request for an empty letterhead, so it falls back to the default.
    """
    return os.environ.get("PRACTICE_NAME", "").strip() or DEFAULT_PRACTICE_NAME


PRACTICE_NAME = practice_name()
